agent/engine/task_manager.go

```diff
@@ -489,8 +494,27 @@ func (mtask *managedTask) cleanupTask(taskStoppedDuration time.Duration) {
 		cleanupTimeBool <- true
 		close(cleanupTimeBool)
 	}()
+	// wait for the cleanup time to elapse, signalled by cleanupTimeBool
 	for !mtask.waitEvent(cleanupTimeBool) {
 	}
+	stoppedSentBool := make(chan bool)
+	go func() {
```

Comment:
This is the desired behavior.  A successful submission of task state will result in this goroutine exiting.  Unsuccessful submissions will delay until success or the timeout or 72 hours, whichever is sooner.  There is no use-case to stop the goroutine other than this.